docs/type-declarations.md

```diff
@@ -1,20 +1,18 @@
 ---
-title: Writing Declarations
+title: Type Declarations
 ---
 
-The real power of the transpiler is unlocked when combining it with declarations for your target environment. Declarations tell TypeScript which Lua API is available in your target context.
+Using TypeScript instead of Lua is useful because everything plugs together in a verifiable way. With that in mind, `tstl` is not very useful unless you pair it with type declarations for your particular Lua environment. That way, TypeScript can catch the typos when you call some API function or use some API variable.
 
-If you need tips or help writing declarations, feel free to [join our Discord](https://discord.gg/BWAq58Y).
+If you need help writing declarations, feel free to [join our Discord server](https://discord.gg/BWAq58Y).
 
 ## About Declaration Files
 
-Declaration files end with the extension _.d.ts_. These contain pure ambient code.
+Declaration files end with the extension `.d.ts` (which stands for "declaration TypeScript file"). Declaration files are different from normal `.ts` files in that they must only contain _ambient_ code. In the context of TypeScript, _ambient_ refers to code that is wiped away by the compiler and not emitted into the actual program output.
```

Comment:
Please don't call it 'wiped away'. Ambient code refers to code that only exists when compiling, but does not appear in the output code emitted by the compilation.